### ISSUE TITLE
Different output on validation between arelleCmdLine and API?

### ISSUE THREAD

**Issue by Felixaverlant** [Arelle/Arelle#1085]
### What happened?

Using the same xbrl instance, I get different output with the 3 different methods : 
- cmdline & GUI: I get a list of errors/warning 
- using the API I only get a None output.

arelle-release : 2.23.7
cmdline version : a779ce3cbe419ad9a9754fad49cc89f5626c38ea

When using the cmdline, it seems to work properly since I get around 80 errors with : 
`python arelleCmdLine.py -f  /PATH/FILE.xhtml -v` 

When using the API, I only get a None output with this code : 
```python
  xcntrl = Cntlr.Cntlr(logFileName="logToBuffer")
  xbrl = xcntrl.modelManager.load(path)
  validation = xbrl.modelManager.validate() 
  pp(validation)
  xbrl.close()
```

Tried to use a wrong file path but it does not give me the same output and I do have facts when I inspect the xbrl model.

Not sure where to dig from here...

### If running from the command line, what command did you run?

python arelleCmdLine.py -f  /PATH/file.xhtml -v

### Interface

CLI (arelleCmdLine), GUI (arelleGUI), Python library (pip install)

### Version

a779ce3cbe419ad9a9754fad49cc89f5626c38ea

### Download

From source (git checkout), pip install

### Operating System

macOS 13 Ventura

**Comment by austinmatherne-wk** [comment]
Hi @Felixaverlant,

Validation results [aren't returned by the validate function](https://arelle.readthedocs.io/en/2.23.9/apidocs/arelle/arelle.ModelManager.html#arelle.ModelManager.ModelManager.validate), they're in the Arelle log buffer.

```python
xcntrl.logHandler.getJson()
# or
xcntrl.logHandler.getXml()
# or
xcntrl.logHandler.getHtml()
# or
xcntrl.logHandler.getText()
```

**Comment by Felixaverlant** [comment]
Thanks for 

> Hi @Felixaverlant,
> 
> Validation results [aren't returned by the validate function](https://arelle.readthedocs.io/en/2.23.9/apidocs/arelle/arelle.ModelManager.html#arelle.ModelManager.ModelManager.validate), they're in the Arelle log buffer.
> 
> ```python
> xcntrl.logHandler.getJson()
> # or
> xcntrl.logHandler.getXml()
> # or
> xcntrl.logHandler.getHtml()
> # or
> xcntrl.logHandler.getText()
> ```

Hi @austinmatherne-wk , thanks for your fast reply.  Makes sens.

Don't know if it was pseudocode but it does not work.
getJson() (and the other ones) are not available in autocomplete. 
Still tried it but get the following error when using getJson() with `logToBuffer`
`object has no attribute 'formulaOptions'`
I looked into the source code and tried different logFileName but get a different message with `logToPrint` :
`AttributeError: 'LogToPrintHandler' object has no attribute 'getJson'`

Sorry JS dev here so unfamiliar with python.

**Comment by austinmatherne-wk** [comment]
Can you provide the full error? Depending on the taxonomy you're loading, FormulaOptions might need to be [set on the ModelManager](https://github.com/Arelle/Arelle/blob/f7a591c1620539f24c8b9c01fb5c6c0a2adc311d/arelle/CntlrCmdLine.py#L889).

**Comment by Felixaverlant** [comment]
Hi,

sure. 

Here's the log: 
```json
{'log': [{'code': 'exception:AttributeError',
          'level': 'error',
          'message': {'error': "'ModelManager' object has no attribute "
                               "'formulaOptions'",
                      'instance': 'efrag-20251231.xhtml',
                      'text': '2024-02-22 09:32:55,473 '
                              '[exception:AttributeError] Instance validation '
                              "exception: 'ModelManager' object has no "
                              "attribute 'formulaOptions', instance: "
                              'efrag-20251231.xhtml - efrag-20251231.xhtml \n'
                              'Traceback (most recent call last):\n'
                              '  File '
                              '"/Users/felixaverlant/Documents/projects/sustainsoft/xbrl/csrd-taxonomy-python/venv/lib/python3.12/site-packages/arelle/Validate.py", '
                              'line 109, in validate\n'
                              '    self.instValidator.validate(self.modelXbrl, '
                              'self.modelXbrl.modelManager.formulaOptions.typedParameters(self.modelXbrl.prefixedNamespaces))\n'
                              '                                                '
                              '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n'
                              "AttributeError: 'ModelManager' object has no "
                              "attribute 'formulaOptions'"},
          'refs': [{'href': 'efrag-20251231.xhtml'}]}]}
```

The source code : 
```python
xcntrl = Cntlr.Cntlr(logFileName='logToBuffer')
xbrl = xcntrl.modelManager.load(finalPath)
validation = xbrl.modelManager.validate() 
logs = json.loads(xcntrl.logHandler.getJson())
pp(logs)
xbrl.close()
```
And the file I'm trying to validate.
[efrag-20251231.xhtml.zip](https://github.com/Arelle/Arelle/files/14371057/efrag-20251231.xhtml.zip)

Thanks again !

**Comment by austinmatherne-wk** [comment]
That taxonomy contains formula assertions. You can try setting `xcntrl.modelManager.formulaOptions = FormulaOptions()`. I'd recommend looking at the link from my previous message to see how the cli code configures formula validation.

**Comment by Felixaverlant** [comment]
Hi Austin, I read `CntlrCmdLine.py` and both FormulaOptions are the same (which seems to be the default:
```json
{
    "cacheSize": 10000000,
    "compileOnly": false,
    "errorUnsatisfiedAssertions": false,
    "formulaAction": null,
    "parameterValues": {},
    "runIDs": null,
    "testcaseResultOptions": null,
    "testcaseResultsCaptureWarnings": false,
    "timeVariableSetEvaluation": false,
    "traceAssertionResultCounts": false,
    "traceCallExpressionCode": false,
    "traceCallExpressionEvaluation": false,
    "traceCallExpressionResult": false,
    "traceCallExpressionSource": false,
    "traceFormulaRules": false,
    "traceParameterExpressionResult": false,
    "traceParameterInputValue": false,
    "traceSatisfiedAssertions": false,
    "traceUnmessagedUnsatisfiedAssertions": false,
    "traceUnsatisfiedAssertions": false,
    "traceVariableExpressionCode": false,
    "traceVariableExpressionEvaluation": false,
    "traceVariableExpressionResult": false,
    "traceVariableExpressionSource": false,
    "traceVariableFilterWinnowing": false,
    "traceVariableFiltersResult": false,
    "traceVariableSetExpressionCode": false,
    "traceVariableSetExpressionEvaluation": false,
    "traceVariableSetExpressionResult": false,
    "traceVariableSetExpressionSource": false,
    "traceVariablesDependencies": false,
    "traceVariablesOrder": false
}
``` 

There's no error raise and the output remains once `FormulaOptions()` set:
```json
{'log': [{'code': '',
          'level': 'info',
          'message': {'text': '2024-03-14 13:55:48,957 [] Formula xpath2 '
                              'grammar initialized in 0.41 secs - \n'},
          'refs': [{'href': ''}]},
         {'code': 'exception:AssertionError',
          'level': 'error',
          'message': {'error': '',
                      'instance': 'efrag-20251231.xhtml',
                      'text': '2024-03-14 13:55:49,268 '
                              '[exception:AssertionError] Instance validation '
                              'exception: , instance: efrag-20251231.xhtml - '
                              'efrag-20251231.xhtml \n'},
          'refs': [{'href': 'efrag-20251231.xhtml'}]}]}
```

I tried to use ValidateXbrl.ValidateXbrl, Validate.validate and the method of the modelManager but the 3 methods gave me the same output.

Is there a way to get more logs ?

Thanks.

**Comment by austinmatherne-wk** [comment]
Hi @Felixaverlant,

What do you get in the logs for the following?
This interface in `CntlrCmdLine` is subject to change, but we'll make something similar available directly from the `Cntlr` module in #1097.
```python
from arelle import CntlrCmdLine
from arelle.RuntimeOptions import RuntimeOptions


if __name__ == '__main__':
    options = RuntimeOptions(
        entrypointFile='https://github.com/Arelle/Arelle/files/14371057/efrag-20251231.xhtml.zip',
        logFile='logToBuffer',
        validate=True,
    )
    cntlr = CntlrCmdLine.configAndRunCntlr(options, None)
    logs = cntlr.logHandler.getText()
```
